File: apps/cargo-tracker/packs/booking/app/domain/booking/domain/cargo_repository.rb
# frozen_string_literal: true

module Booking
  module Domain
    # 貨物予約リポジトリの出力ポート（ヘキサゴナルの抽象契約）。
    class CargoRepository
      def save(_cargo)
        raise NotImplementedError, "#{self.class}#save is not implemented"
      end

      def find_by_booking_id(_booking_id)
        raise NotImplementedError, "#{self.class}#find_by_booking_id is not implemented"
      end

      def find_by_tracking_number(_tracking_number)
        raise NotImplementedError, "#{self.class}#find_by_tracking_number is not implemented"
      end

      def all(*)
        raise NotImplementedError, "#{self.class}#all is not implemented"
      end
    end
  end
end

File: apps/cargo-tracker/src/contexts/booking/domain/repository/cargo-repository.ts
import type { Cargo } from '../model/cargo.js';

/** 貨物リポジトリ出力ポート */
export interface CargoRepository {
  save(cargo: Cargo): Promise<number>;
  findByBookingId(bookingId: string): Promise<Cargo | null>;
  update(cargo: Cargo): Promise<void>;
}
